Ingress/controllers/nginx-third-party/nginx/manager.go

```diff
@@ -0,0 +1,168 @@
+	sslProtocols        = "TLSv1 TLSv1.1 TLSv1.2"
+	workerProcesses     = "auto"
+	sslSessionTimeout   = "10m"
+	sslSessionCacheSize = "10m"
+	sslDirectory        = "/etc/nginx-ssl"
+)
+
+// defaultConfiguration returns the default configuration contained
+// in the file default-conf.json
+func newDefaultNginxCfg() nginxConfiguration {
+	return nginxConfiguration{
+		BodySize:                 bodySize,
+		DefaultTimeout:           120,
+		EnforceHTTPS:             false,
+		ErrorLogLevel:            errorLevel,
+		HtsIncludeSubdomains:     true,
+		HtsMaxAge:                htsMaxAge,
+		KeepAlive:                75,
+		MaxWorkerConnections:     768,
+		ProxyConnectTimeout:      30,
+		ProxyRealIPCIDR:          defIPCIDR,
+		ProxyReadTimeout:         30,
+		ProxySendTimeout:         30,
+		ProxyWriteTimeout:        30,
+		ServerNameHashMaxSize:    512,
+		ServerNameHashBucketSize: 64,
+		SSLBufferSize:            sslBufferSize,
+		SSLCiphers:               sslCiphers,
+		SSLProtocols:             sslProtocols,
+		SSLSessionCache:          true,
+		SSLSessionCacheSize:      sslSessionCacheSize,
+		SSLSessionTickets:        true,
+		SSLSessionTimeout:        sslSessionTimeout,
+		UseProxyProtocol:         false,
+		UseGzip:                  true,
+		WorkerProcesses:          workerProcesses,
+	}
+}
+
+func newNginxCfg(kubeClient *unversioned.Client, defaultSvc, customErrorSvc, tcpServices string) *templatedNginx {
+	cfg := templatedNginx{
+		defBackend:      getService(kubeClient, defaultSvc),
+		defResolver:     skydnsIPPort(kubeClient),
+		defError:        getService(kubeClient, customErrorSvc),
+		servicesL4:      getTcpServices(kubeClient, tcpServices),
+		sslDHParam:      ssl.SearchDHParamFile(sslDirectory),
+		sslCertificates: ssl.CreateSSLCerts(sslDirectory),
+	}
+
+	if len(cfg.servicesL4) == 0 {
+		glog.Info("no tcp service defined. skipping")
+	} else {
+		glog.Infof("tcp service/s defined: %v", cfg.servicesL4)
+	}
+
+	return &cfg
+}
+
+// SyncRequired determines if it is required to reload nginx or not checking
+// changes in ingress rules and custom nginx configuration
+func (ngx *NginxManager) SyncRequired(ingresses *extensions.IngressList) bool {
+	if ngx.reloadRequired {
+		return true
+	}
+
+	if reflect.DeepEqual(ngx.IngressList.Items, ingresses.Items) {
```

Comment:
You're only able to do this because we don't update status yet. when we do, we should avoid reloading nginx for a status update. Just compare spec? 

You can ignore the case where the controller updates the Status with nodeIP:80 and someone overwrite it (through kubectl). Ideally a controller should overwrite that overwrite, becauase only controllers should be setting status, but I'm fine with just a todo. Of course if you really want to handle that case you can too, by doing a field compare before you decide to send/not-send the status update (but decouple that from reloading nginx).